packages/grafana-ui/src/components/VizLegend/VizLegendTableItem.tsx

```diff
@@ -24,26 +25,69 @@ export const LegendTableItem: React.FunctionComponent<Props> = ({
   className,
 }) => {
   const styles = useStyles(getStyles);
+  const { eventBus } = useContext(PanelContext);
+
+  const onMouseEnter = useCallback(
+    (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+      if (eventBus) {
+        eventBus.publish({
+          type: DataHoverEvent.type,
+          payload: {
+            raw: event,
+            x: 0,
+            y: 0,
+            dataId: item.label,
+          },
+        });
+      }
+    },
+    [eventBus, item]
+  );
+
+  const onMouseOut = useCallback(
+    (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+      if (eventBus) {
+        eventBus.publish({
+          type: DataHoverClearEvent.type,
+          payload: {
+            raw: event,
+            x: 0,
+            y: 0,
+            dataId: item.label,
+          },
+        });
+      }
+    },
+    [eventBus, item]
+  );
+
+  const onClick = useCallback(
+    (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+      if (onLabelClick) {
+        onLabelClick(item, event);
+      }
+    },
+    [item, onLabelClick]
+  );
+
+  const onColorChange = useCallback(
+    (color: string) => {
+      if (onSeriesColorChange) {
+        onSeriesColorChange(item.label, color);
+      }
+    },
+    [item, onSeriesColorChange]
+  );
```

Comment:
This entire code seems to be shared between VizLegendTableItem and VizLegendListItem. Also, it feels wrong the event's but is retrieved so deep in the structure. I think you could get the context in the VizLegend and pass the callbacks to VizLegendTable and VizLegendList.

Alternatively - but this is a way bigger decision yet to be made - maybe we don't need the callbacks approach at all, but instead introduce a new event types, i.e. SeriesColorChange and subscribe to such an event in the EventBus and do the config modifications? Something to think about gping forward. For this PR let's stick with the callbacks!